pkg/ingester/index/index.go

```diff
@@ -244,6 +244,11 @@ func (shard *indexShard) delete(labels labels.Labels, fp model.Fingerprint) {
 		j := sort.Search(len(fingerprints.fps), func(i int) bool {
 			return fingerprints.fps[i] >= fp
 		})
+
+		// see if search didn't find fp which matches the condition which means we don't have to do anything.
+		if j < len(fingerprints.fps) && fingerprints.fps[j] == fp {
```

Comment:
It seems to me that this logic is now inverted... if this condition is true, we should delete. What am I missing?

Reply:
Ahh,right. Sorry, my bad. Will fix it.

Reply:
Great thanks. I'm actually a bit surprised it wasn't caught by a test. Would it make sense to cover it by a unit test?